hometasks/2-building-blocks-of-oop-part-1/src/Triangle.ts

```diff
@@ -0,0 +1,58 @@
+import { Shape } from './Shape';
+import { Point } from './Point';
+
+export class Triangle extends Shape {
+    constructor(point1: Point, point2: Point, point3: Point);
+    constructor(point1: Point, point2: Point, point3: Point, color: string, filled: boolean);
+    constructor(point1: Point, point2: Point, point3: Point, color?: string, filled?: boolean) {
+        
+        if(!point1 || !point2 || !point3) {
+            throw new Error(`Points number is not correct`);
+        }
+
+        const points: Point[] = [];
+        points.push(point1);
+        points.push(point2);
+        points.push(point3);
+
+        if (color && filled) {
+            super(points, color, filled);
+        } else {
+            super(points);
+        }
+        
+    }
+
+    public toString(): string {
+        let objString = `Triangle[`
+        this.points.map( (point, index, arr) => {
+            objString = objString + `v${index+1}=(${point.getX}, ${point.getY})`;
+            if(index < arr.length - 1) {
+                objString += ',';
+            } else {
+                objString += ']';
+            }
+        });
+
+        return objString;
+    }
+
+    public getType(): string {
+        
+        const equalityLimit = 0.01;
```

Comment:
this should be the part of sides calculation and not when comparing the sides...